builtin/logical/pki/backend.go

```diff
@@ -377,3 +406,134 @@ func (b *backend) invalidate(ctx context.Context, key string) {
 func (b *backend) periodicFunc(ctx context.Context, request *logical.Request) error {
 	return b.crlBuilder.rebuildIfForced(ctx, b, request)
 }
+
+func (b *backend) initializeStoredCertificateCounts(ctx context.Context) error {
+	b.tidyStatusLock.Lock()
+	defer b.tidyStatusLock.Unlock()
+	// For performance reasons, we can't lock on issuance/storage of certs until a list operation completes,
+	// but we want to limit possible miscounts / double-counts to over-counting, so we take the tidy lock which
+	// prevents (most) deletions
+
+	entries, err := b.storage.List(ctx, "certs/")
+	if err != nil {
+		return err
+	}
+	atomic.StoreUint32(b.certCount, uint32(len(entries)))
+	metrics.SetGauge([]string{"secrets", "pki", "total_certificates_stored"}, float32(*b.certCount))
```

Comment:
Would it not be better to move both calls to SetGauge till after the for loops with the appropriate values so they only appear on the metrics with the "correct value"?

Reply:
I'll let others weigh in here, but wanted to outline my thinking:
* This would only ever be "high" (a 'correct' value for our contract)
* "Fixing" this would require far more code complexity (another backend-bool measure, or another set of decrement calls), since the decrement calls below also report metrics.
* Sorting long lists could take a very, very long time, in precisely the case we would want these metrics "immediately"

Reply:
Fixing this - I've add it to the tidy-status call with a warning that deduplication isn't complete.